### DISCUSSION TITLE
Limit data viewable by users (Hide phonenumber for example)

### DISCUSSION THREAD

**Discussion by GewoonJaap** [pocketbase/pocketbase#1328] [Q&A]
Hi,

I was wondering if it is possible to create rules/hooks to hide certain data from collections.
For example I want my users to be able to get basic data from other users, like username, signup date etc. However I don't want users accessing each others phonenumber or other private details that maybe present in the User model.

My question: How to restrict users from reading that private data, but it should be possible to read their own private data ;)

Note: I don't want to filter those fields out in the front-end, would be kinda unsecure ;)

**Comment by ganigeorgiev** [comment]
You can move your private fields in their own collection and associate them to your users with a single (aka. Max Select = 1) `relation` field. The access to those fields could be restricted using the API rules of the collection.

To fetch a user with their "profile" data you can use `expand` (direct or indirect, depending on the type of relation you choose) - https://pocketbase.io/docs/expanding-relations/.

**Reply by GewoonJaap** [reply]
Ah okay, that could be possible indeed. Is this best practice for now? Because when I am writing my own APIs I would probably write a query on the user table and exclude some fields (server side).

But I am new to this so I don't know if that would be possible and if it is good practice. I can definitely give your solution a try

**Reply by ganigeorgiev** [reply]
> Is this best practice for now?

There is no other way at the moment, unless [you use PocketBase as framework](https://pocketbase.io/docs/use-as-framework/) and write your own [custom query](https://pocketbase.io/docs/database/).

**Reply by GewoonJaap** [reply]
If I use PocketBase as framework, will I still have the default functionality available? (The web UI and all the tools that come with it) Because if that is the case, then I would happily try out the framework to extend functionality 👍

**Reply by ganigeorgiev** [reply]
> If I use PocketBase as framework, will I still have the default functionality available? 

Yes, the standard functionality is included and you can extend/overwrite most of the default behavior via [event hooks](https://pocketbase.io/docs/event-hooks/).
The prebuilt executable uses this example go file - https://github.com/pocketbase/pocketbase/blob/master/examples/base/main.go.

But for this specific use case it is more involving and there are a lot of edge cases in relation to the "hidden" fields because you have a user model response from several different places, it is not just 1 endpoint (eg. after create, update, authentication, etc.) and you probably would want some conditional behavior (eg. only the owner and admin to have access to the hidden fields), so it may be easier to just move your "hidden" fields in a `profiles` collection with its own API rules and constraints.

**Reply by GewoonJaap** [reply]
Ah okay, great!

Yeah indeed, I will probably stick to your solution. But I was also looking into some ratelimiting or captcha solutions. So might try build that.
